pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts

```diff
@@ -0,0 +1,232 @@
+import {Filters} from "../queryFilter";
+import {nodeRegionsByIDSelector} from "../store/nodes";
+import {selectIsTenant} from "../store/uiConfig";
+import {
+  selectDatabaseDetails,
+  selectDatabaseDetailsGrantsSortSetting,
+  selectDatabaseDetailsTablesFiltersSetting,
+  selectDatabaseDetailsTablesSearchSetting,
+  selectDatabaseDetailsTablesSortSetting,
+  selectDatabaseDetailsViewModeSetting
+} from "../store/databaseDetails/databaseDetails.selectors";
+import {
+  combineLoadingErrors,
+  generateTableID,
+  getNodesByRegionString,
+  normalizePrivileges,
+  normalizeRoles
+} from "../util";
+import {selectTableDetails} from "../store/databaseTableDetails/tableDetails.selector";
+
+const selectTables = (state: AppState, props: RouteComponentProps): DatabaseDetailsPageDataTable[] => {
+  const database = getMatchParamByName(props.match, databaseNameAttr);
+  const dbDetails = selectDatabaseDetails(state);
+  const tableDetails = selectTableDetails(state);
+  const dbTables = dbDetails.cache[database]?.data?.results.tablesResp.tables;
+
+  return dbTables.map(table => {
+    const tableId = generateTableID(database, table);
+    const details = tableDetails.cache[tableId];
+    const roles = normalizeRoles(details?.data?.results.grantsResp.grants.map(grant => grant["user"]));
+    const grants = normalizePrivileges([].concat(...details?.data?.results.grantsResp.grants.map(grant => grant["privileges"])));
+    const nodes = details?.data?.results.stats.replicaData.nodeIDs || [];
+    // Query fetches distinct table indexes.
+    const numIndexes = details?.data?.results.schemaDetails.indexes.length;
+    return {
+      name: table,
+      loading: !!details?.inFlight,
+      loaded: !!details?.valid,
+      lastError: details?.lastError,
+      details: {
+        columnCount:
+          details?.data?.results.schemaDetails.columns?.length || 0,
+        indexCount: numIndexes,
+        userCount: roles.length,
+        roles: roles,
+        grants: grants,
+        statsLastUpdated:
+          details?.data?.results.heuristicsDetails
+            .stats_last_created_at || null,
+        hasIndexRecommendations:
+          details?.data?.results.stats.indexStats
+            .has_index_recommendations || false,
+        totalBytes:
+          details?.data?.results.stats.spanStats.total_bytes || 0,
+        liveBytes: details?.data?.results.stats.spanStats.live_bytes || 0,
+        livePercentage:
+          details?.data?.results.stats.spanStats.live_percentage || 0,
+        replicationSizeInBytes:
+          details?.data?.results.stats.spanStats.approximate_disk_bytes ||
+          0,
+        nodes: nodes,
+        rangeCount:
+          details?.data?.results.stats.spanStats.range_count || 0,
+        nodesByRegionString: getNodesByRegionString(
+          nodes,
+          nodeRegionsByIDSelector(state),
+          selectIsTenant(state),
+        ),
+      },
+    };
+  });
+};
+
+export const mapStateToProps = createSelector(
```

Comment:
Is there a reason why this is created as a selector? I believe we talked about why this is unnecessary for the other cases.